Fixes #951: setMagicka and setFatigue have no effect

lgromanowski opened the pull request:
Only change magicka, or fatigue when one of affecting attributes changed.

Signed-off-by: Lukasz Gromanowski lgromanowski@gmail.com

lgromanowski committed:
Fixes #951: setMagicka and setFatigue have no effect

Only change magicka, or fatigue when one of affecting attributes changed.

Signed-off-by: Lukasz Gromanowski <lgromanowski@gmail.com>

ghost commented:
No, this will not work. You're just checking if the modifier isn't 0, rather than checking if the modifier or the base changed.

lgromanowski committed:
Fixes #951: setMagicka and setFatigue have no effect

Check if there is difference between current and previous attribute
modifiers related to fatigue and magica, if there is then update
corresponding stat.

Signed-off-by: Lukasz Gromanowski <lgromanowski@gmail.com>

lgromanowski commented:
It seems that @scrawl fixed this in https://github.com/scrawl/openmw/commit/2a8ab932efcd6c05887146561a44f6338799e21c so I'm closing this.
